modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/PageMemoryMvPartitionStorage.java

```diff
@@ -72,8 +83,17 @@ public PageMemoryMvPartitionStorage(int partitionId, TableView tableConfig, Page
 
         groupId = StorageUtils.groupId(tableConfig);
 
-        versionChainFreeList = ((VolatilePageMemoryDataRegion) dataRegion).versionChainFreeList();
-        rowVersionFreeList = ((VolatilePageMemoryDataRegion) dataRegion).rowVersionFreeList();
+        try {
+            versionChainFreeList = createVersionChainFreeList(dataRegion.pageMemory(), groupId, partitionId);
```

Comment:
Why did you decide to create these lists per partition? It would be better to turn them into parameters. Maybe there was a slight misunderstanding between us -  volatile region should have shared freelists to better utilize memory, it's an optimization that we should do. Persistent regions must have separate freelists for each partition. Right now we have separate lists in volatile region, which is not what I expect.

Reply:
Ok, I think I see what you mean. Applied the change.